src/client.c

```diff
@@ -1705,22 +1727,38 @@ static gboolean add_recipe_host(const gchar *value, AppData *app_data,
     gchar *connect_uri = NULL;
     gchar *id = NULL;
     gboolean result = TRUE;
+    gchar **ssh_destination;
 
     args = g_strsplit(value, "=", 2);
     if (g_strv_length(args) != 2) {
         id = g_strdup_printf("%u", (*recipe_id)++);
-        connect_uri = g_strdup (args[0]);
+        connect_uri = args[0];
     } else {
-        id = g_strdup(args[0]);
-        connect_uri = g_strdup (args[1]);
+        id = args[0];
+        connect_uri = args[1];
     }
 
-    RecipeData *recipe_data = new_recipe_data(app_data, id);
-    recipe_data->connect_uri = connect_uri;
-    recipe_data->rhost = parse_host(recipe_data->connect_uri);
+    ssh_destination = parse_host (connect_uri);
+
+    if (ssh_destination != NULL) {
+        RecipeData *recipe_data;
+
+        recipe_data = new_recipe_data (app_data, id);
+        recipe_data->connect_uri = g_strjoinv (NULL, ssh_destination);
+        recipe_data->rhost = ssh_destination[1];
+
+        g_free (ssh_destination[0]);
+        g_free (ssh_destination);
+
+        result = TRUE;
+    } else {
+        result = FALSE;
+    }
 
     g_free (id);
-    g_strfreev (args);
+    g_free (connect_uri);
+    g_free (args);
```

Comment:
According to glib doc, you should be using g_strfreev to free data allocated with g_strsplit.

Reply:
From `g_strsplit()`,
> Returns a newly-allocated NULL-terminated array of strings. Use g_strfreev() to free it.

There is nothing in `g_strsplit()` requiring `g_strfreev()`, it's only for convenience.

From g_strfreev(),
> Frees a NULL-terminated array of strings, as well as each string it contains.

We are using the "strings" allocated by `g_strsplit()` in `id` and `connect_uri`, and we are freeing them ourselves. We only need to free the array of strings.

Reply:
Since we're consecutively freeing id, connect_uri and args, why not just use g_strfreev()?

Reply:
```
args = g_strsplit(value, "=", 2);
if (g_strv_length(args) != 2) {
    id = g_strdup_printf("%u", (*recipe_id)++);
    connect_uri = args[0];
```
If `args` has 1 string, `connect_uri` is `args[0]`, and we make `id` from `recpie_id`.

```
} else {
    id = args[0];
    connect_uri = args[1];
}
```
If `args` has 2 strings, `id` is `args[0]` and `connect_uri` is `args[1]`.

To use `g_strfreev()`, we'd need to make condition to check where `id` is pointing to. Or, duplicate the strings in `args` just for the sake of using `g_strfreev ()`, which is pointless.

Instead of duplicating the strings in `args`, we are using them directly in `id` and `connect_uri`, and we don't need extra conditions to free `id`, because that was handled already in the assignment.

Reply:
I see what the issue is.  Thank you for the explanation.